src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseGradingService.java

```diff
@@ -115,6 +119,58 @@ public ProgrammingExerciseGradingService(ProgrammingExerciseTestCaseService test
                 // This method will return without triggering the build if the submission is not of type TEST.
                 triggerTemplateBuildIfTestCasesChanged(programmingExercise.getId(), result.getId());
             }
+
+            if (!isSolutionParticipation && !isTemplateParticipation) {
+
+                // check if a manual result exists, if so we want to clone and update this with the new updated test-case and sca feedback
+                Optional<ProgrammingExerciseStudentParticipation> studentParticipation = programmingExerciseParticipationService
+                        .findStudentParticipationWithLatestManualResultsAndFeedbacksAndRelatedSubmissionsAndAssessor(participation.getId());
+
+                if (studentParticipation.isPresent() && studentParticipation.get().getResults().stream().findFirst().isPresent()) {
+
+                    Result manualResult = studentParticipation.get().getResults().stream().findFirst().get();
```

Comment:
I think we should use the latest result, not the first when iterating through a set. It is not impossible that a participation has two manual results (be it later because of the second correction). Could we use `findLatestResult()` here?

Reply:
The repository method here is
```
findStudentParticipationWithLatestManualResultsAndFeedbacksAndRelatedSubmissionsAndAssessor
```
but the one in the changes above is 
```
findByIdWithLatestManualResultAndFeedbacksAndRelatedSubmissionsAndAssessor
```
In particular, the first one mentions "Result***s***" and the latter only "Result". Is this something we need to worry about?

Reply:
There is only ever one result returned from the repository method.

Reply:
I will rename this to singular.